addons/medical_engine/functions/fnc_handleDamage.sqf

```diff
@@ -88,42 +88,48 @@ if (_hitPoint isEqualTo "ace_hdbracket") exitWith {
         _receivedDamage = _damageStructural;
         _woundedHitPoint = "Body";
     };
-    TRACE_2("received",_receivedDamage,_woundedHitPoint);
 
     // Environmental damage sources all have empty ammo string
     // No explicit source given, we infer from differences between them
     if (_ammo isEqualTo "") then {
-        // Fall damage and being hit by vehicle always have a shooter
-        // Check this first because fire damage could happen while falling
+        // Any collision with terrain/vehicle/object has a shooter
+        // Check this first because burning can happen at any velocity
         if !(isNull _shooter) then {
-            // Shooter can be _unit if hit by empty vehicle so check velocity too
-            // A piloted vehicle could hit falling unit so still check shooter
+            _ammo = "#collision";
+
+            /*
+              If shooter != unit then they hit unit, otherwise it could be:
+               - Unit hitting anything at speed
+               - An empty vehicle hitting unit
+               - A physX object hitting unit
+               Assume fall damage for downward velocity because it's most common
+            */
             if (_shooter == _unit && {(velocity _unit select 2) < -2}) then {
-                // Low damage indicates < ~5m fall
-                _woundedHitPoint =  if (_receivedDamage < 0.35) then {
-                    selectRandom ["LeftLeg", "RightLeg"];
-                } else {
-                    selectRandom ["LeftLeg", "RightLeg", "Body", "Head"];
-                };
-                _ammo = "#falling";
-                TRACE_5("Fall Damage",_unit,_shooter,_instigator,_damage,_receivedDamage);
+                _woundedHitPoint = selectRandom ["RightLeg", "LeftLeg"];
+                TRACE_5("Fall",_unit,_shooter,_instigator,_damage,_receivedDamage);
             } else {
-                _woundedHitPoint = "Body";
-                _ammo = "#vehiclecrash";
-                TRACE_5("Collision Damage",_unit,_shooter,_instigator,_damage,_receivedDamage);
+                _woundedHitPoint = selectRandom ["RightArm", "LeftArm", "RightLeg", "LeftLeg"];
+                TRACE_5("Collision",_unit,_shooter,_instigator,_damage,_receivedDamage);
+            };
+
+            // Significant damage suggests high relative velocity
+            // Momentum transfers to body/head for worse wounding
+            if (_receivedDamage > 0.35) then {
+                 _woundedHitPoint = selectRandom ["Body", "Head"];
```

Comment:
I feel like we should actually raise an additional wounding event in these cases (if you hit the ground at high speed the injury wouldn't fully transfer to your body - you'd still break your legs).